# 04.Karyotype
This folder is used for ancestral karyotype analysis.

The abbreviations in the document are represented the species name, which is found in [species.txt](https://github.com/Dived-Jin/Gecko_Sexchromosome/blob/af3026331796bd45a6fcf1bb4dc1415eb5a400ed/species.txt).

## (1) a.DESCHRAMBLER.sh
The command is used to identify the synteny fragment in all species. 
```
used: sh a.DESCHRAMBLER.sh
  - divergence_tree: the species divergence time tree with NWK format, generated by 03phylogeny/c.divergence.sh.
  - paremater_file: the parameter file for DESCHRAMBLER, such as example/DESCHRAMBLER/params.500k.txt
  - Makefile: the parameter file for DESCHRAMBLER, such as example/DESCHRAMBLER/Makefile.SFs
  - config: the parameter file for DESCHRAMBLER, such as example/DESCHRAMBLER/config.SFS
```
## (2) b.anges.sh:
Constructed the karyotype for each node. The control file in example/ANGES
```
  used: sh b.anges.sh
```
## (3) c.merge_karyotype.sh
The command merges ANGES and plots the karyotypeã€‚
```
used: sh c.merge_karyotype.sh <hubpath> <conversion_segment>
  - hubpath: the result of ANGES. 
  - conversion_segment: the DESCHRAMBLER result.
```

## (4) d.grimm.sh
the variant of karyotype between two nodes.
```
used: sh d.grimm.sh <Ancestral_karyotype_merge> <node1> <node2>
  - Ancestral_karyotype_merge: the karyotype file was generated by  c.merge_karyotype.sh.
  - node1 and node2: the id of the node which needs compare.
```
