#/bin/bash
#######bin path
scriptp=`dirname "$(realpath $0)"`
binpath="$scriptp/bin"
pairixpath="" #https://github.com/4dn-dcic/pairix.git
hicexplorerpath="" #https://github.com/deeptools/HiCExplorer.git
############# Raw input
HiCalignment=$1  # the merged_nodups.txt file is generated by HiC read aligned to genome by juicer # https://github.com/aidenlab/juicer.git
genomefasta=$2
spnname=$3

############step1 merge alignment to 4dn-pairs converter 
samtools faidx $genomefasta 
awk '{print $1"\t"$2}' $genomefasta.fai |sort -k2,2n >$spnname.sizes
export PATH="$pairixpath/bin:$PATH"
perl $pairixpath/util/merged_nodup2pairs.pl $HiCalignment $spnname.sizes $spnname

###########step2
##100kb 
$hicexplorerpath/bin/cooler cload pairix $spnname.sizes:100000 $spnname.bsorted.pairs.gz $spnname.bsorted.pairs.gz.100kb.cool
hicConvertFormat --matrices $spnname.bsorted.pairs.gz.100kb.cool --outFileName $spnname.bsorted.pairs.gz.100kb.cool.h5 --inputFormat cool --outputFormat h5 --load_raw_values
hicCorrectMatrix correct --matrix $spnname.bsorted.pairs.gz.100kb.cool.h5 --correctionMethod KR -o $spnname.bsorted.pairs.gz.100kb.cool.KR.h5
hicConvertFormat --matrices $spnname.bsorted.pairs.gz.100kb.cool.KR.h5 --outFileName $spnname.bsorted.pairs.gz.100kb.cool.KR.hicpro --inputFormat h5 --outputFormat hicpro --bedFileHicpro $spnname.bsorted.pairs.gz.100kb.cool.KR.hicpro.100kb.bed
